src/node_api.cc

```diff
@@ -40,6 +40,11 @@ class BufferFinalizer : private Finalizer {
         static_cast<BufferFinalizer*>(hint)};
     finalizer->_finalize_data = data;
 
+    // FIXME(gabrielschulhof): We need a `v8::HandleScope` when we retrieve the
+    // `node::Environment` here because the finalizer may not have been called
+    // by way of JavaScript. Remove if/once we have
+    // https://chromium-review.googlesource.com/c/v8/v8/+/1879902
+    v8::HandleScope(finalizer->_env->isolate);
```

Comment:
```suggestion
    v8::HandleScope handle_scope(finalizer->_env->isolate);
```

Otherwise the object gets destroyed immediately and there’s no actual handle scope :)

Reply:
🤦